### PR TITLE
Simplify estimator train_and_eval.

### PR DESCRIPTION
**Pull request by mingxingtan**
After asking TF team, looks like tf.estimator.train_and_eval is recommended for continuous train_and_eval, as it avoids rebuild graph for every train.

### CODE REVIEW

**Comment by fsx950223** [comment]
lgtm

**Review by NikZak — COMMENTED** [review]
It's definitely more straightforward and it's the correct usage so we should adopt this change. I saw in the issues (#719) that there is memory leak in pure train mode so this change might bring back the memory leak. But we may try solve it from there

**Comment by mingxingtan** [comment]
@NikZak If the OOM still exists, we can add back the multiprocessing option. Could you help give it a test?

File: efficientdet/main.py [hunk 1]
```diff
@@ -254,56 +243,36 @@ def _can_partition(spatial_dim):
...
+      FLAGS.validation_file_pattern,
+      is_training=False,
+      use_fake_data=FLAGS.use_fake_data,
+      max_instances_per_image=max_instances_per_image)
 
   # start train/eval flow.
   if FLAGS.mode == 'train':
-    total_examples = int(config.num_epochs * FLAGS.num_examples_per_epoch)
-    _train(total_examples // FLAGS.train_batch_size)
+    total_examples = int(config.num_epochs * FLAGS.num_examples_per_epoch),
```
Comment (fsx950223) [inline efficientdet/main.py:272]: Could I remove the line?

Comment (mingxingtan) [reply efficientdet/main.py:272]: Oh, yes, this line should be deleted!
I saw you already fixed in the other PR. Thanks for the fix!
